video/out/wayland_common.c

```diff
@@ -1778,9 +1766,9 @@ static void set_geometry(struct vo_wayland_state *wl, bool resize)
     vo_calc_window_geometry2(vo, &screenrc, wl->scaling, &geo);
     vo_apply_window_geometry(vo, &geo);
 
-    greatest_common_divisor(wl, vo->dwidth, vo->dheight);
-    wl->reduced_width = vo->dwidth / wl->gcd;
-    wl->reduced_height = vo->dheight / wl->gcd;
+    int gcd = greatest_common_divisor(vo->dwidth, vo->dheight);
```

Comment:
You would need to check which one of these is greater.
```suggestion
    int gcd = greatest_common_divisor(MPMAX(vo->dwidth, vo->dheight), MPMIN(vo->dwidth, vo->dheight));
```
Or however you want to do it.

Reply:
`b > a => a % b = a` which means `gcd(b, a)` automatically falls through to `gcd(a, b)`

Reply:
Ah right.